scripts/tests/twister_blackbox/test_printouts.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Blackbox tests for twister's command line functions
+"""
+
+import importlib
+import mock
+import os
+import pytest
+import sys
+
+
+class TestPrintOuts:
+    ZEPHYR_BASE = os.getenv("ZEPHYR_BASE")
+
+    TESTDATA_1 = [
+        (
+            os.path.join(ZEPHYR_BASE, 'tests', 'kernel'),
+            ['mailbox', 'apic_tsc', 'timer', 'linker_generator', 'libc',
+             'cache', 'security', 'scheduler', 'multi_heap', 'isr_table',
+             'kernel', 'fpu', 'demand_paging', 'fifo', 'threads', 'interrupt',
+             'memory slabs', 'userspace', 'memory_protection', 'smp', 'sleep',
+             'pm', 'memory_heap', 'mmu', 'condition_variables', 'message queue',
+             'spinlock', 'memory_slabs', 'workqueue', 'base', 'mutex', 'heap',
+             'xip', 'device', 'picolibc']
+        ),
+        (
+            os.path.join(ZEPHYR_BASE, 'tests', 'posix'),
+            ['newlib', 'picolibc', 'posix', 'eventfd', 'filesystem', 'net',
+             'getopt', 'socket']
+        ),
+    ]
+
+    TESTDATA_2 = [
+        (
+            os.path.join(ZEPHYR_BASE, 'tests', 'posix', 'getopt'),
+            [
+                'portability.posix.getopt.getopt',
```

Comment:
It is going to be a problem using live test data to verify twister itself, i.e. if someone adds or changes a test, the testsuite will fail. This might not be detected in the same CI run and will show up only later when those tests are run over the modified data.

I suggest to do this using some dummy tests that are maintained as part of the testsuite itself rather than relying on actual test data.